x-pack/legacy/plugins/uptime/server/lib/domains/license.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { RequestHandlerContext } from 'kibana/server';
+
+export type UMLicenseCheck = (context: RequestHandlerContext) => boolean;
+
+export const licenseCheck: UMLicenseCheck = ({ licensing: { license } }) => {
+  if (license === null) {
+    throw Boom.badRequest('Missing license information');
+  }
+  if (!license.isOneOf(['basic', 'standard', 'gold', 'platinum', 'trial'])) {
+    throw Boom.forbidden('License not supported');
+  }
+  if (license.isActive === false) {
+    throw Boom.forbidden('License not active');
+  }
+  return true;
```

Comment:
Boom errors are not properly handled by NP by default (it will always returns a 500). don't forget to wrap handlers that may returns boom errors with `router.handleLegacyErrors`. Note that this will be deprecated after 8.0, and that plugin should manually returns 'valid' errors using  the request handler's response object.

Reply:
I've done away with the `Boom` throw entirely